src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -639,18 +638,19 @@ def _visit_map(self, node: gtir.FunCall) -> ValueExpr:
 
         # GT4Py guarantees that all connectivities used to generate lists of neighbors
         # have the same length, that is the same value of 'max_neighbors'.
-        assert 1 == len(
+        local_offset_providers = dace_utils.filter_connectivities(
             {
-                self.subgraph_builder.get_offset_provider(offset).max_neighbors  # type: ignore[union-attr]
+                offset: self.subgraph_builder.get_offset_provider(offset)
                 for offset in input_local_offsets
             }
         )
-        local_offset = input_local_offsets[0]
-        offset_provider = self.subgraph_builder.get_offset_provider(local_offset)
-        assert isinstance(offset_provider, gtx_common.Connectivity)
+        if len(set(table.max_neighbors for table in local_offset_providers.values())) != 1:
+            raise ValueError(
+                "Unexpected arguments to map expression with different local dimensions."
+            )
+        local_offset, offset_provider = next(iter(local_offset_providers.items()))
```

Comment:
Bellow on line 679 you override this values.
I think it is confusing to create and define a variable outside a loop and then overwrite (not update or modify) it inside a loop.
You should rewrite it such that you do not have this.
The same is true for `local_offset`.

This is actually a feature I like in rust that everything is constant except you explicitly mark it as mutual.
And that you are forced to write curly brackets in `if` expressions.

Reply:
You will easily convince me by just mentioning Rust 🦀 

I agree, overwriting a variable at some point in a loop is hard to follow, in any programming language. In this case, it is even not needed.